sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs

```diff
@@ -544,6 +545,45 @@ public void DuplicateKeysInLogRecordAttributesAndLogScope2()
             Assert.Equal(duplicateScopeValue2, actualScopeValue);
         }
 
+        [Fact]
+        public void ValidateCustomEvent()
+        {
+            // Arrange.
+            var logRecords = new List<LogRecord>(1);
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.AddInMemoryExporter(logRecords);
+                });
+            });
+
+            var eventLogger = new CustomEventLogger(loggerFactory);
+
+            var attributesList = new List<KeyValuePair<string, object?>>
+            {
+                new KeyValuePair<string, object?>("customEventKey", "customEventValue")
+            };
+
+            eventLogger.TrackEvent("MyCustomEvent", attributesList);
+
+            // Assert.
+            var logRecord = logRecords.Single();
+            var logRecodBatch = new Batch<LogRecord>(new[] { logRecord } , 1);
+            var properties = new ChangeTrackingDictionary<string, string>();
+            var telemetryItems = LogsHelper.OtelToAzureMonitorLogs(logRecodBatch, null, "Ikey");
+
+            var telemetryItem = telemetryItems.FirstOrDefault();
+            var telemetryEventData = telemetryItem?.Data.BaseData as TelemetryEventData;
```

Comment:
Using `FirstOrDefault` doesn't help here and may make it harder to troubleshoot test failures.

```suggestion
            var telemetryItem = telemetryItems.Single();
            var telemetryEventData = telemetryItem.Data.BaseData as TelemetryEventData;
```